frontend/pages/blocks/index.js

```diff
@@ -1,24 +1,69 @@
 import Head from "next/head";
 
+import { useContext, useEffect } from "react";
+
+import BlocksApi from "../../components/api/Blocks";
+import { DataContext } from "../../components/utils/DataProvider";
+
 import Header from "../../components/Header";
 import Blocks from "../../components/Blocks";
 import Footer from "../../components/Footer";
 
 import "bootstrap/dist/css/bootstrap.min.css";
 
-export default () => (
-  <div>
-    <Head>
-      <link rel="shortcut icon" type="image/png" href="/static/favicon.ico" />
-      <title>Near Explorer | Blocks</title>
-    </Head>
-    <Header />
-    <Blocks />
-    <Footer />
-    <style jsx global>{`
-      body {
-        background-color: #f8f8f8;
-      }
-    `}</style>
-  </div>
-);
+const Index = ({ blocks, total, start, stop }) => {
+  const ctx = useContext(DataContext);
+
+  useEffect(() => {
+    ctx.setBlocks(blocks);
+    ctx.setPagination(pagination => {
+      return {
+        ...pagination,
+        total,
+        start,
+        stop
+      };
+    });
+    console.log("set");
+  }, []);
+
+  return (
+    <div>
+      <Head>
+        <link rel="shortcut icon" type="image/png" href="/static/favicon.ico" />
+        <title>Near Explorer | Blocks</title>
+      </Head>
+      <Header />
+      <Blocks />
+      <Footer />
+      <style jsx global>{`
+        body {
+          background-color: #f8f8f8;
+        }
+      `}</style>
+    </div>
+  );
+};
+
+Index.getInitialProps = async () => {
+  let start = 0;
+  let stop = 0;
+  let total = 0;
+  let blocks = [];
+
+  try {
+    blocks = await BlocksApi.getLatestBlocksInfo();
+    total = await BlocksApi.getTotal();
+
+    if (blocks.length > 0) {
+      start = blocks[0].height;
+      stop = blocks[blocks.length - 1].height;
+    }
+  } catch (err) {
+    console.log(err);
```

Comment:
It would be quite problematic to find the place where this err gets printed if we need to. Let's have a practice at least provide some useful information in the message.